app/components/Modal/ConfirmModal.tsx

```diff
@@ -63,130 +60,71 @@ type WithModalProps = {
-        onCancel = () => Promise.resolve(),
-        message,
-        title,
-        closeOnCancel = true,
-        closeOnConfirm = false,
-        ...props
-      } = this.props;
+  children: (props: { openConfirmModal: () => void }) => ReactNode;
 
-      const wrapAction = (
-        action: () => Promise<any>,
-        closeWhenDone: boolean
-      ) => {
-        return () => {
-          const onResolve = closeWhenDone
-            ? (result) => {
-                this.stopWorking();
-                this.hideModal();
-                this.resetError();
-                return result;
-              }
-            : (result) => result;
+  // The following props are only passed on to ConfirmModalContent
+  message: ReactNode;
+  title: string;
+  cancelText?: string;
+  confirmText?: string;
+  danger?: boolean;
+};
 
-          const onError = (error) => {
-            this.stopWorking();
-            const errorMessage =
-              get(error, ['meta', 'errorMessage']) || 'Det skjedde en feil...';
-            this.setErrorMessage(errorMessage);
-            throw error;
-          };
+export const ConfirmModal = ({
+  onConfirm = async () => {},
+  onCancel = async () => {},
+  closeOnCancel = true,
+  closeOnConfirm = false,
+  children,
+  ...contentProps
+}: ConfirmModalProps) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [working, setWorking] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string>();
 
-          this.resetError();
-          this.startWorking();
-          return action().then(onResolve, onError);
-        };
-      };
+  const wrapAction = <T,>(action: () => Promise<T>, closeWhenDone: boolean) => {
+    return async () => {
+      setErrorMessage(undefined);
+      setWorking(true);
 
-      const modalOnConfirm = wrapAction(onConfirm, closeOnConfirm);
-      const modalOnCancel = wrapAction(onCancel, closeOnCancel);
-      const { working, errorMessage } = this.state;
-      return (
-        <>
-          <WrappedComponent
-            {...(props as Record<string, any>)}
-            onClick={this.toggleModal}
-          />
-          <Modal
-            closeOnBackdropClick={!working}
-            show={this.state.modalVisible}
-            onHide={this.toggleModal}
-          >
-            <ConfirmModal
-              onCancel={modalOnCancel}
-              onConfirm={modalOnConfirm}
-              message={message}
-              title={title}
-              disabled={working}
-              errorMessage={errorMessage}
-            />
-          </Modal>
-        </>
-      );
-    }
+      try {
+        const result = await action();
+        if (closeWhenDone) {
+          setWorking(false);
+          setModalVisible(false);
+          setErrorMessage(undefined);
+        }
+        return result;
+      } catch (error) {
+        setWorking(false);
+        setErrorMessage(
+          (error as any)?.meta?.errorMessage || 'Det skjedde en feil...'
```

Comment:
hmmm, not a huge fan :|

(Also, wont this give a TS error with strict mode?)

Reply:
Yes I know, I hate it too. Do you know what type we are expecting here? I suspect it might be an API-error of some sort, but I didn't want to spend too much time looking into it

Reply:
Yeah that is what it looks like. Could you make it generic that contains the expected parameters?

Reply:
Out of scope